modules/tinymce/src/core/main/ts/api/NotificationManager.ts

```diff
@@ -95,12 +95,17 @@ function NotificationManager(editor: Editor): NotificationManager {
     });
   };
 
-  const open = function (spec: NotificationSpec) {
+  const open = function (spec: NotificationSpec, fireEvent: boolean = true) {
     // Never open notification if editor has been removed.
     if (editor.removed || !EditorView.isEditorAttachedToDom(editor)) {
       return;
     }
 
+    // fire event to allow notification spec to be mutated before display
+    if (fireEvent) {
```

Comment:
Is this the right place for this? I'm wondering if this should be just before the `getImplementation().open()` call, as at the moment this will fire, but a notification may not open if the text, etc... is the same due to the `isEqual` checks below.

Reply:
Fair point. Do we want to allow modification of the notification so that it can be made to pass those checks?

Reply:
Honestly, I don't know. I can see pros and cons to both sides, so just thought I'd comment to see what others thought.

At this stage though, I'd say leave it where it is as showing the same notification twice is more of an edge case.

Reply:
~If there's two notifications that would be the same normally but the integrator changes one of them (whether that's the text, icon, or whatever), it'd fail `isEqual` and show both, right? Idk why an integrator would only change one though, since I can't think of any way to use the `spec` to usefully identify notifications e.g. triggered by different things but with the same text, etc.~

~As for the event firing for duplicate notifications... Yeah, it might be weird if that happens then~

Just realised, what if someone uses this to log notifications to a server or something, as like an error reporting thing? I know Atl do that kind of thing. If the event fires for notifications that don't end up showing, especially for dupe notifications, wouldn't that be annoying? That's the best use case I can think of for moving it, since nothing about changing the spec seems too bad.